### ISSUE TITLE
Avalonia - Issue tracker

### ISSUE THREAD

**Issue by MutantAura** [Ryujinx/Ryujinx#3662]
## Bug Report

The Avalonia GUI currently has a few bugs and issues ranging from minor to major. This issue serves as a tracker and easily extendable way to follow progress on specific limitations. The tracker will be broken into "Major", "Moderate" and "Minor" which range from bugs that need to be resolved before Avalonia can become default all the way to more 'nice-to-have' features.

Leave a comment if there is anything **reproducible** that has been missed. This is also not an issue to add feature requests to. Comments asking for new GUI features will be marked as off-topic.

### Major

- [x] Updater does not function. Creates a blank dialog box with no content. (Fixed in https://github.com/Ryujinx/Ryujinx/pull/3670)
- [x] Using Vulkan on Windows (have checked Linux on Nvidia and Intel, does not occur there) the program is extremely unresponsive and takes disproportionate time to load GUI elements compared to OpenGL (Menus, tooltips, changing between windows etc.). Appears to have started after the merge of https://github.com/Ryujinx/Ryujinx/pull/3483. (Fixed by https://github.com/Ryujinx/Ryujinx/pull/3674)
- [x] Overlays cause serious issues especially when entering fullscreen. The Nvidia overlay will glitch for multiple seconds and shadowplay becomes nearly unusable. RTSS and similar monitoring software will also monitor the UI itself rather than the game window, making them also functionless. (Fixed in https://github.com/Ryujinx/Ryujinx/pull/3674)
- [x] Some Linux distros show a transparent render window using Vulkan. This also extends to any dialog boxes.
- [x] ContentDialog aren't visible in fullscreen on macOs (https://github.com/Ryujinx/Ryujinx/issues/4939)
### Moderate

- [x] Type boxes do not lose focus when clicking emptyspace and will remain active until an element that has a focus property is selected. Whitespace does not appear to have any focus properties. (Fixed in https://github.com/Ryujinx/Ryujinx/pull/3991)
![image](https://user-images.githubusercontent.com/44103205/194761090-1a68ba5b-794d-492b-ab73-e96113d63f29.png)
![image](https://user-images.githubusercontent.com/44103205/194761283-ece7b65b-c238-4d26-8471-bea716f7b3c4.png)
- [x] "Hide Cursor on Idle" setting does not work when hovering the render window on Windows 10/11. (Pending: https://github.com/Ryujinx/Ryujinx/pull/4266)
- [x] Entering a letter into an AutoCompleteTextBox will cause it to populate the entire screen vertically. (Fixed in https://github.com/Ryujinx/Ryujinx/pull/3643)
- [x] Fullscreening using Vulkan can cause the window to crash and become unresponsive. (Fixed in: https://github.com/Ryujinx/Ryujinx/pull/3483)
- [x] Launching a game via terminal will crash when using Vulkan. (Fixed in https://github.com/Ryujinx/Ryujinx/pull/3674)
- [x] Frame-pacing can be problematic compared to GTK. This is incredibly hard to verify though given all the usual tools (as listed above) don't reliably work as they capture the UI, not the game directly. (Fixed in https://github.com/Ryujinx/Ryujinx/pull/3674)
- [x] Fullscreening prevents the creation of ControllerApplet dialogs. This issue also continues after exiting fullscreen. (Possible fix: https://github.com/Ryujinx/Ryujinx/pull/4663)
- [x] NCA extraction dialogs do not close once task is completed (Fixed by: https://github.com/Ryujinx/Ryujinx/pull/4569)
- [x] Software keyboard text does not fill with example text provided by title (Fixed by https://github.com/Ryujinx/Ryujinx/pull/4671).
- [x] Hide UI use F4 hotkey, but if you enable it, there is no way to get the UI back by pressing F4.

### Minor

- [x] Using arrowkeys in an AutoCompleteTextBox will select the first entry and remove all other options. Fixed by https://github.com/Ryujinx/Ryujinx/pull/3752
- [x] Locales are far from complete. Will likely need to switch to a translation service to fully mitigate this issue. (CrowdIn integration)
- [x] Toggling fullscreen on Windows causes the window size to slowly increase in vertically. (Image after around 8 toggles)
![image](https://user-images.githubusercontent.com/44103205/188010477-066666ac-c0d7-4322-a9ef-549f96da7990.png)
- [x] Clicking text on menuitems should toggle the checkbox in the menuitem. (Framework limitation, workaround in https://github.com/Ryujinx/Ryujinx/pull/3469)
- [x] Volume status bar is misaligned (Partially resolved in https://github.com/Ryujinx/Ryujinx/pull/3643)
- [ ] Text font and navbar/button icons could scale with the "icon size" slider as some users may appreciate larger GUI fonts.
- [x] Window does not remember it's size position or state on relaunch (Fixed by: https://github.com/Ryujinx/Ryujinx/pull/4657)

### Currently unfixable by Ryujinx/Framework bug

- [ ] Scaling on non-standard DPI may not function correctly. A better way to query DPI is coming in Avalonia 11.0
- [x] XDG Portal support is missing for Flatpack build (Apparently fixed by Avalonia 11.0)
- [ ] Menu doesn't show on Steam Deck when launched via Steam (This also happens to the official devtools of Avalonia) (Fixed by: https://github.com/AvaloniaUI/Avalonia/pull/14366)

### Environment?

- Ryujinx version: Avalonia - Various

### Additional context?

List originally from discord: https://discord.com/channels/410208534861447168/814263228926132274/996196878590943243 but the message length was getting to its limit.

**Comment by Daisouji** [comment]
- The Program Window doesn't stay maximized, when u do restart the program (Win 10)

**Comment by emmauss** [comment]
The overlay issue, unfortunately can't be fixed. The window is what's being rendered and not just the game view. The game view is part of the window, and not a child window like in other toolkits

**Comment by MutantAura** [comment]
> The overlay issue, unfortunately can't be fixed. The window is what's being rendered and not just the game view. The game view is part of the window, and not a child window like in other toolkits

That's a really unfortunate issue because it has a bit of a knock-on effect beyond just overlays. Game capture in OBS for instance will also capture the GUI and filtering software like reshade etc impact the UI elements too. Could a potential solution not simply be to have an option to pop-out an SDL window instead of an embedded window at game launch? (similar to Dolphin I guess. Image for reference)
![image](https://user-images.githubusercontent.com/44103205/188309336-a62d7c31-4fd7-4c66-8494-ac2bfa9213f5.png)

**Comment by emmauss** [comment]
> > The overlay issue, unfortunately can't be fixed. The window is what's being rendered and not just the game view. The game view is part of the window, and not a child window like in other toolkits
> 
> That's a really unfortunate issue because it has a bit of a knock-on effect beyond just overlays. Game capture in OBS for instance will also capture the GUI and filtering software like reshade etc impact the UI elements too. Could a potential solution not simply be to have an option to pop-out an SDL window instead of an embedded window at game launch? (similar to Dolphin I guess. Image for reference) ![image](https://user-images.githubusercontent.com/44103205/188309336-a62d7c31-4fd7-4c66-8494-ac2bfa9213f5.png)

A pop out can be made,  but it needs to be investigated. The main window still controls a lot of things when the game is running. A lot of the functionality would need to be made free of the main window.

**Comment by gdkchan** [comment]
Can't you just embed a separate window for rendering rather than the separate window or the current approach where everything is rendered on the same window? I think think a separate window or those compromises would be a downgrade from the GTK UI.

**Comment by emmauss** [comment]
> Can't you just embed a separate window for rendering rather than the separate window or the current approach where everything is rendered on the same window? I think think a separate window or those compromises would be a downgrade from the GTK UI.

Embedding a window doesn't work well on windows. The embedded window still receives all input messages, including focus. so focus can easily be grabbed by either the main window or the embedded window, preventing input from going to the other window. If the main window gets focus, the game window won't have input and vise versa. for linux, input is still passed to the parent before going to the child, so focus doesn't matter.

**Comment by Squall-Leonhart** [comment]
not sure if its expected, but Avalonia builds are also slightly slower when decoding intros.

**Comment by LunaDook** [comment]
Noticing in the Polish translation at least, the "Start Game in Fullscreen Mode" button has some weird rendering issues. On one computer with a 1440p monitor at 125% resolution scaling, the last word in the line disappears entirely:

![image](https://user-images.githubusercontent.com/7586972/192178453-647b7a82-e619-48d5-abb8-68d46f0240bd.png)

On a system with a 4K display at 200% scaling, the line has the last word, but the last letter is cut off (it's supposed to be an m):

![image](https://user-images.githubusercontent.com/7586972/192178565-38a9dd6c-6dc7-4413-bd23-e3aa2306941f.png)

Both systems are running the same build of Ryujinx.

**Comment by ShadowLady** [comment]
When doing "Manage DLC" on a game with multiple DLCs (Xenoblade C2 on my case) and trying to scroll, the window expands instead.

**Comment by MetrosexualGarbodor** [comment]
Should this be kept open for the remaining, albeit minor issues?

**Comment by Squall-Leonhart** [comment]
the on request window to configure a controller when none is connected isn't opening, and the log is just looping requests for a controller.

**Comment by OldManKain** [comment]
**Hide Curser on Idle** doesn't work on game rendering window. It does work on loading shader, bottom status and top drop-down bars.

**Comment by EmulationFanatic** [comment]
The error applet seems to display a code related error as opposed to the firmware-provided error message.
![image](https://user-images.githubusercontent.com/62343878/204154649-20620940-e43c-4361-8ab8-94bec6e60541.png)

**Comment by gdkchan** [comment]
> The error applet seems to display a code related error as opposed to the firmware-provided error message. ![image](https://user-images.githubusercontent.com/62343878/204154649-20620940-e43c-4361-8ab8-94bec6e60541.png)

That was already fixed on #3888 no?

**Comment by EmulationFanatic** [comment]
Totally missed that PR. Sorry!

**Comment by IsaacMarovitz** [comment]
> I would say yes, until it's default we may find more things as we go. Plus I personally want the typebox issue looked at 🙃

Fixed in #3991

**Comment by gdkchan** [comment]
When you set a "Graphics shader dump path", Avalonia shows a prompt asking if you want to disable it when the game is started. However, clicking on "Yes" does nothing. The path is still set and it asks it every time you launch the game, even if you click on "Yes".

**Comment by Pixxel123** [comment]
On Steam Deck, it seems that Avalonia does not show dropdown menus on click or touch input in the top settings bar. I have opened #4073 to track this in greater detail.

**Comment by OldManKain** [comment]
Something I noticed today since I got a new controller and wanted to do some testing.

If a controller becomes disconnected during emulation the emulator freezes with the console window spamming that there is no controller detected, when you reconnect the controller the console stops spamming but the emulator stays frozen.

**Controllers Tested**
Official Switch Pro 
Official Joy-Con
Xbox Series
Xbox 360
Chrome Series Xbox 360
GuliiKit KingKong 2 Pro
GulliKit KingKong 2

**Comment by joshuacode751** [comment]
Saving settings while in game. The controller stop's working. The controller works again when saving the settings on the Input section.

**Comment by GamerzHell9137** [comment]
Some issues found on Avalonia builds

1. If an update is available and you deny updating the emulator, the option to update it afterwards will be unavailable

https://user-images.githubusercontent.com/2002038/211300980-4e340b25-b1ba-49f3-8178-a5cee332109e.mp4

2. If emulation was started and stopped the updater afterwards will not be able to be triggered

https://user-images.githubusercontent.com/2002038/211301464-8c595fe4-bed7-4a34-baa7-31e46ae112fc.mp4

3. Cheats, Manage User Profiles and About window can't be closed with ESC button

**Comment by IsaacMarovitz** [comment]
<h1>Nitpicks:</h1>

<h2>Issues caused by Avalonia:</h2>

- Menu items have a weird space gap between the shortcut and edge of item
<img width="260" alt="Screenshot 2023-01-11 at 13 41 37" src="https://user-images.githubusercontent.com/42140194/211890996-2077a00b-4a97-4ce2-a14d-b68f82a4fb42.png">

<h2>Issues caused by FluentAvalonia:</h2>

- These controls overlap
<img width="216" alt="Screenshot 2023-01-11 at 13 43 43" src="https://user-images.githubusercontent.com/42140194/211891423-a8364069-c5b3-46b6-ad9f-318c4784a791.png">
- These controls are janky
<img width="612" alt="Screenshot 2023-01-11 at 13 44 45" src="https://user-images.githubusercontent.com/42140194/211891625-5b754e38-8c9d-4b9b-9ef1-36b14fa60512.png">

<h2>Issues caused by Ryujinx:</h2>

- Settings input menu is janky **(Fix in Progress)**
- Settings window border isn't consistent with rest of app
- Fix some 'GTK-ness' in Update, DLC, and Cheat dialogs
- Inconsistencies in Settings window
<img width="436" alt="Screenshot 2023-01-11 at 13 47 54" src="https://user-images.githubusercontent.com/42140194/211892293-abe5fdb7-aa6d-40bb-951c-0672e47bac1f.png">
<img width="189" alt="Screenshot 2023-01-11 at 13 48 05" src="https://user-images.githubusercontent.com/42140194/211892324-b7b79954-c74d-44a7-90a2-b9951b0d39c8.png">
- Standardise styling and colours
- Create debug window with examples of controls and standards

**Comment by GamerzHell9137** [comment]
### Nitpick:

Audio changes are spammed on the console log

Suggestion is to make it only spam once when the handle is let off.

![image](https://user-images.githubusercontent.com/2002038/211967013-1837eebd-6eb5-4afc-9215-26e99457406a.png)

**Comment by gm1224** [comment]
When extracting the exefs, romfs, or logos of a game, the progress dialog doesn't automatically close once the success dialog appears.

![image](https://user-images.githubusercontent.com/86170792/212585682-42d2d0b2-b9bd-4111-ada7-197e405ab49b.png)

**Comment by IsaacMarovitz** [comment]
> When extracting the exefs, romfs, or logos of a game, the progress dialog doesn't automatically close once the success dialog appears. Both windows must be closed manually.

Will be fixed by #4275

**Comment by AcK77** [comment]
> Will be fixed by https://github.com/Ryujinx/Ryujinx/pull/4275

Nop, this fix is still WIP, my PR only implement a way to send Notification popup and cleanup some different things.

**Comment by EliEron** [comment]
I just noticed that the Avalonia Software keyboard does not populate the input with a default value when the game provides one. Like the suggested name in Fire Emblem and the like:

GTK:
![Ryujinx_rAaQddZ2aZ](https://user-images.githubusercontent.com/22575945/213172607-752172e4-b458-43bf-aad1-edf3c9527ef6.jpg)

Avalonia:
![Ryujinx Ava_5XAJoy2N62](https://user-images.githubusercontent.com/22575945/213172626-be18022e-99b6-4775-aa6f-78ecbc77a78a.jpg)


These are both screenshots from master.

**Comment by OldManKain** [comment]
**Can't change Time Zone**

Found a Regression that wasn't noticed until today when someone said they couldn't change their time zone.

1.1.513.
![image](https://user-images.githubusercontent.com/47503099/223927072-3a6c1f06-9ad2-4b99-ac52-9f07485b8630.png)


1.1.512
![image](https://user-images.githubusercontent.com/47503099/223926931-cb367c29-4c42-4d1c-8db2-54a906ae7c4f.png)

1.1.513
![image](https://user-images.githubusercontent.com/47503099/223927425-1e87f032-f0f3-48f7-991f-57cdff9f4389.png)

**Comment by piplup55** [comment]
ava ui seems to system date even when you change it in the application, https://youtu.be/OgzWbJWNIqc

**Comment by Neristee** [comment]
### Crash when using the mouse to go into fullscreen (last tested with Ryujinx.ava 1.1.777)

**Steps to reproduce :** 

- Launch a game in windowed mode
- Use the UI to click Options then Fullscreen (didn't test using F11)
- Ryujinx.ava and the Ryujinx Console both crash and disappear

The issue, albeit seemingly random, appears quite frequently. Enough to annoy me during testing some PR on _BOTW_ and _Metroid Prime Remaster_. I was able to reproduce it with 1.1.777 on _Pokemon Violet_ twice in 12 min (as the logs provided in this comment show).

[Ryujinx_1.1.777_2023-05-11_12-08-18.log](https://github.com/Ryujinx/Ryujinx/files/11456612/Ryujinx_1.1.777_2023-05-11_12-08-18.log)
[Ryujinx_1.1.777_2023-05-11_12-20-35.log](https://github.com/Ryujinx/Ryujinx/files/11456613/Ryujinx_1.1.777_2023-05-11_12-20-35.log)

These two logs are link to the two configurations here : 
![first](https://github.com/Ryujinx/Ryujinx/assets/15640409/2e84782e-9170-463b-9328-849330f9475b)
![second](https://github.com/Ryujinx/Ryujinx/assets/15640409/e8f4b1d5-a2c5-4190-b3e3-dc04385f48e5)

I though it was a Direct Mouse Input issue so I disabled it. Didn't change much as the logs both show the same thing.

Don't hesitate to ping me if more testing is needed.

**Comment by Khobztajine** [comment]
Hi, F4 won't hide UI, I have to manually remove UI with action bar and Hide UI, F4 can only show UI but not hide it. Thanks

**Comment by Foul-Tarnished** [comment]
Not really a bug, but

The audio volume bar should get closed when you go back into fullscreen.

For now it just stays open, and you can see it (and interact with it) in fullscreen
but cannot close it until you go back to windowed, click somewhere, then go back to fullscreen

**Comment by NONONO1V** [comment]
It seems that in the settings interface, options can only be set in the order of user interface -> input -> keyboard hotkeys -> ..., and the order cannot be changed, otherwise the game will crash.



https://github.com/Ryujinx/Ryujinx/assets/109849776/4ea715a5-0575-47d8-a86b-1c1d2af5e7d6

[Ryujinx_1.1.826_2023-05-28_12-06-32.log](https://github.com/Ryujinx/Ryujinx/files/11583649/Ryujinx_1.1.826_2023-05-28_12-06-32.log)

**Comment by noxifoxi** [comment]
# Application Icon in Linux issue
Application Icon in Linux is shifted/wrapped and low quality

| Avalonia  | GTK |
| ------------- | ------------- |
| ![image](https://github.com/Ryujinx/Ryujinx/assets/293639/2696fb25-a2c2-4477-8dfd-7452950b8949) | ![image](https://github.com/Ryujinx/Ryujinx/assets/293639/48e0277d-314d-4c12-9eac-b1ecaec0b898)
| ![image](https://github.com/Ryujinx/Ryujinx/assets/293639/b8bbbfe1-d753-4557-a520-0631188a83c0) | ![image](https://github.com/Ryujinx/Ryujinx/assets/293639/aa6479b4-8335-4eae-81f7-724e147c5db1)
| ![image](https://github.com/Ryujinx/Ryujinx/assets/293639/f0f8a73d-1d0e-4ca2-93de-ab3d792c40a5) | ![image](https://github.com/Ryujinx/Ryujinx/assets/293639/f701063d-4d78-440b-955b-d8056edf1385)

**Comment by tarrouye** [comment]
Input bugs: 

1. Mouse/Touch input goes through Avalonia menus and is consumed by the application

2. Mouse/Touch input never releases, preventing any application input until emulation is restarted. 

![image](https://github.com/Ryujinx/Ryujinx/assets/5017270/c5c28fa6-5613-425d-a2a3-f131de6ece3b)
 https://youtu.be/9-_l1KV88pM

**Comment by SamusAranX** [comment]
The game list (or at least the bindings for Time Played and Last Played) should be reloaded on game exit or language change. 

As of right now, those two values won't update without a manual list refresh.

**Comment by tarrouye** [comment]
https://github.com/Ryujinx/Ryujinx/assets/5017270/30838177-5f0f-45e7-af89-ec146e720f55

Inputs are repeated/delayed/out of order

**Comment by AcK77** [comment]
> Inputs are repeated/delayed/out of order. A little hard to demonstrate, but pretty easy to feel for yourself if you just type enough.

As far as I remember this is an expected behavior to avoid refreshing the GameList each time you hit a key to search a game, there is a little delay to let you type the full word and then refresh the GameList using the search filter.

**Comment by tarrouye** [comment]
> As far as I remember this is an expected behavior to avoid refreshing the GameList each time you hit a key to search a game, there is a little delay to let you type the full word and then refresh the GameList using the search filter.

Sure, it makes sense to have a debouncer for the search functionality. However, could you maybe point to where it is implemented? Because there seems to be a bug, a debouncer should not cause repeated or out of order inputs

**Comment by kevincrans** [comment]
I have 5 bugs on 1.1.859 M2 Apple silicon Ventura, correct me if I made a mistake:

1. Configuring input, enter presses closes the settings menu.
2. The touchscreen doesn't respond to the mouse.
3. Rarely the up/down keys are mapped to the menubar.
4. Rarely keys are not working, but give feedback sound.
5. When playing in fullscreen, the menubar is black, this is because the menubar hasn't integrated with the macOS menubar yet.

At least the first 2 bugs weren't present in 1.1.0-macos1 build, I don't think 4 either.

**Comment by kevincrans** [comment]
6. I can't use E for Input (ZL to be specific)

Is input Avalonia related?

**Comment by notjl** [comment]
I found a bug when playing ToTK 1.1.2 with [Playstation UI Mod](https://gamebanana.com/mods/443201). It will crash and return this error line in the log:
```
00:00:28.461 |E| Application : Unhandled exception caught: System.InvalidOperationException: Unable to create child window for native control host. Application manifest with supported OS list might be required.
   at Avalonia.Win32.Win32NativeControlHost.DumbWindow..ctor(Boolean layered, Nullable`1 parent) in /_/src/Windows/Avalonia.Win32/Win32NativeControlHost.cs:line 99
```

Tried playing using the GTK executable, it's working fine with the mod. When using Ava with the mod, it crashes.

**Comment by AcK77** [comment]
A log file could be more useful, thanks.

**Comment by Squall-Leonhart** [comment]
Avalonia does appear to require the OS Support section in the generated app.manifest for some reason for such child windows, 

The manifest currently embedded only sets DPI compatibility.

**Comment by notjl** [comment]
Here's the log file https://hastebin.com/share/vavukodesu.yaml

**Comment by SamusAranX** [comment]
Found two macOS-specific issues:
* Command+Q doesn't immediately quit the program. For some reason it takes several presses for Ryujinx to actually quit
* Command+Comma doesn't open the Settings window. This is less of a bug and more of a missing feature. That key combination is the macOS-standard way of quickly getting to an app's settings.

**Comment by vesper8** [comment]
Avalonia 11 just released! https://github.com/AvaloniaUI/Avalonia/releases/tag/11.0.0

**Comment by AzureStrikerRai** [comment]
I ran into 3 issues while using the Avalonia build.

1. Controller profiles made in GTK do not load. You have to redo them in Avalonia.
2. Loading controller profiles also load the input device you had when saving the profile so it can cause problems when using multiple controllers that use the same controller profile.
3. Using the arrow keys when trying to change deadzones/thresholds either gives the minimum or the maximum. You can't go up by .1 nor .10 by holding ctrl and using the arrow keys like in GTK.

**Comment by IsaacMarovitz** [comment]
Following the update to Avalonia 11 yesterday, the plan to sunset GTK is imminent. We are waiting on the async PRs, which should significantly improve the speediness of many windows, and all the necessary migration infrastructure to get people over from GTK seamlessly.

**Comment by Narugakuruga** [comment]
Updater now will crash in 1.1.900 if the download process lasts for more than 30s

**Comment by GabCoolDude** [comment]
The scrollbar in the about menu looks weird.

Sometimes it goes over names.
![image](https://github.com/Ryujinx/Ryujinx/assets/123309403/e86c5c76-a559-4b5f-a8b0-864ee3e01a55)

And if you scroll down it clips through the bottom.
![image](https://github.com/Ryujinx/Ryujinx/assets/123309403/ccaa7108-b2b6-496e-9c46-21e8a7295556)

**Comment by Narugakuruga** [comment]
> Updater now will crash in 1.1.980 if the download process lasts for more than 30s

This issue was gone in latest version. (Or maybe it can't be reproduced again)

**Comment by rings2006** [comment]
i'm aware there is an issue about it already, but avilonia 11 adds accessibility apis for assistive tech such as screen readers, implimenting that would be nice, and can be seen as an issue to those who need it

**Comment by noxifoxi** [comment]
The issue with the [broken application icon under Linux](https://github.com/Ryujinx/Ryujinx/issues/3662#issuecomment-1567586523) Is fixed (for me).

**Comment by TSRBerry** [comment]
Yep, this was fixed with the update to Avalonia 11 :tada:

**Comment by mubinulhaque** [comment]
`Menu doesn't show on Steam Deck when launched via Steam (This also happens to the official devtools of Avalonia)`

Is this still an issue on Avalonia 11.0? Or has it not been tested yet?

**Comment by Littl3Guy** [comment]
Controls return to default settings when switching between player profiles:

https://github.com/Ryujinx/Ryujinx/assets/115218507/f57f32fd-8cde-4e87-9cc8-70bd3a0bc168

**Comment by willianholtz** [comment]
In Linux, when a message appears (error, warning, etc.), the window often behaves very strangely. Sometimes small, and sometimes full screen!

![Area Grande MK](https://github.com/Ryujinx/Ryujinx/assets/23488207/314e231e-1ea5-4f1a-921d-f601a40d4acc)
![Area pequena](https://github.com/Ryujinx/Ryujinx/assets/23488207/8fcfc696-1690-442c-be45-cc51f35d91cf)

**Comment by nixxou** [comment]
![image](https://github.com/Ryujinx/Ryujinx/assets/45721836/ce485d71-3f3a-4286-894a-56b92f0eb33c)

Nothing much, but when you press Alt There is not shortcut to go to the option menu
There is one for file (F), One for Action (A), but nothing for Options
That's annoying when you want to automate stuff where there is no hotkey availiable and i prefer to rely on keyboard than clicking.

**Comment by nixxou** [comment]
On Windows, when i want to use the Cheat windows on a game, the emulator crash and i have this error : 
00:00:04.086 |E| Application : Unhandled exception caught: System.NullReferenceException: Object reference not set to an instance of an object.
   at Ryujinx.Ava.UI.Views.Main.MainMenuBarView.OpenCheatManagerForCurrentApp(Object sender, RoutedEventArgs e) in D:\a\Ryujinx\Ryujinx\src\Ryujinx.Ava\UI\Views\Main\MainMenuBarView.axaml.cs:line 171

**Comment by willianholtz** [comment]
They could add a button for "default" in the options.

**Comment by urbanracer34** [comment]
Hello,

I don't mean to be a bother, but when will Avalonia be fully merged into the main codebase? We've been sitting on 22/25 tasks completed for a long time now. 

Thanks!

**Comment by MetrosexualGarbodor** [comment]
@urbanracer34 Avalonia was [fully merged  on July 2022](https://github.com/Ryujinx/Ryujinx/pull/3441), it's just not the default GUI on Windows and Linux yet. 
Anyways, [nothing is stopping you from using it as your main build currently](https://github.com/Ryujinx/Ryujinx/wiki/FAQ-and-Troubleshooting#-where-can-i-try-this-avalonia-thing), unless you're on a Steam Deck.

**Comment by urbanracer34** [comment]
> @urbanracer34 Avalonia was [fully merged on July 2022](https://github.com/Ryujinx/Ryujinx/pull/3441), it's just not the default GUI on Windows and Linux yet. Anyways, [nothing is stopping you from using it as your main build currently](https://github.com/Ryujinx/Ryujinx/wiki/FAQ-and-Troubleshooting#-where-can-i-try-this-avalonia-thing), unless you're on a Steam Deck.

Hello! 

Thanks for taking the time to respond, and thanks for the info. 

I want to use it on a Steam Deck. That is the issue I have.

**Comment by urbanracer34** [comment]
Can I please get some help regarding getting this build running on my Steam Deck?

**Comment by DocMAX** [comment]
can i disable this UI and have the old one? gamescope doesn't show the menus with new UI.

**Comment by fvig2001** [comment]
RAM usage is pretty high.
Here's mine on load of application.
<img width="703" alt="RAM usage" src="https://github.com/Ryujinx/Ryujinx/assets/31123547/248df606-7c3d-4e3e-bd70-7a53fbecb5a9">

**Comment by marysaka** [comment]
Gamescope related issues are fixed with https://github.com/AvaloniaUI/Avalonia/pull/14366, waiting on Avalonia side.

**Comment by IsaacMarovitz** [comment]
Now that Avalonia has been made default, Avalonia issues should be like regular issues.
